lib/pdf_attachment_reporter.rb

```diff
@@ -0,0 +1,179 @@
+    def initialize(owning_organisation_name, created_at)
+      @owning_organisation_name = owning_organisation_name
+      @created_at = created_at
+    end
+  end
+
+  def find_pdf_attachment_data(attachment)
+    attachable_object = find_attachable_object(attachment)
+
+    if attachable_object
+      if attachable_object.class == PolicyGroup
+        PDFAttachmentData.new(POLICY_GROUPS, attachment.created_at)
+      else
+        # Responses are only sometimes linked to organisations (via a consultation)
+        if attachable_object.is_a?(Response)
+          if attachable_object.consultation
+            last_published_edition_with_attachment = find_last_published_edition_with_attachment(attachable_object.consultation, attachment.attachment_data)
+
+            if last_published_edition_with_attachment
+              pdf_owning_organisation = guess_organisation_owner_of_edition(last_published_edition_with_attachment)
+              first_published_edition_with_attachment = find_first_published_version_with_attachment(attachable_object.consultation, attachment.attachment_data)
+            end
+          end
+        else
+          last_published_edition_with_attachment = find_last_published_edition_with_attachment(attachable_object, attachment.attachment_data)
+
+          if last_published_edition_with_attachment
+            pdf_owning_organisation = guess_organisation_owner_of_edition(last_published_edition_with_attachment)
+            first_published_edition_with_attachment = find_first_published_version_with_attachment(attachable_object, attachment.attachment_data)
+          end
+        end
+
+        if pdf_owning_organisation && pdf_owning_organisation.live? &&
+            first_published_edition_with_attachment && first_published_edition_with_attachment.public_timestamp
+          PDFAttachmentData.new(pdf_owning_organisation.name, first_published_edition_with_attachment.public_timestamp)
+        end
+      end
+    end
+  end
+
+  def find_attachable_object(attachment)
+    attachable_type = attachment.attachable_type.constantize
+    attachable_type.find_by_id(attachment.attachable_id)
+  end
+
+  def find_last_published_edition_with_attachment(edition, attachment_data)
+    edition.document.editions.order('created_at DESC').detect do |ed|
+      Edition::POST_PUBLICATION_STATES.include?(ed.state) && ed.attachments.any? { |a| a.attachment_data_id == attachment_data.id }
+    end
+  end
+
+  def find_first_published_version_with_attachment(edition, attachment_data)
+    edition.document.editions.order('created_at ASC').detect do |ed|
+      Edition::POST_PUBLICATION_STATES.include?(ed.state) && ed.attachments.any? { |a| a.attachment_data_id == attachment_data.id }
+    end
+  end
+
+  def guess_organisation_owner_of_edition(edition)
+    if edition.organisations.first.class == WorldwideOrganisation
+      # Editions associated with a WWO are associated with one only
+      edition.organisations.first.sponsoring_organisation
```

Comment:
I was confused by this as I thought the organisations association would always be `Organisations` (because of https://github.com/alphagov/whitehall/blob/master/app/models/edition/organisations.rb#L13-L14 and https://github.com/alphagov/whitehall/blob/master/app/models/edition_organisation.rb#L3).  But then I found that corp info pages are the culprit (see: https://github.com/alphagov/whitehall/blob/master/app/models/corporate_information_page.rb#L66-L72).  How frustrating!

I was on the fence about changing the conditional to be about doing something different only if the edition is a corp info page, but then I realised I might not have found all the places where this might be the case, and I don't think it'd make anything clearer because we'd still need the else case if the first org *isn't* a WWO.  Might be worth leaving a comment about corp info pages though, just so future folk don't go through the same voyage of discovery though.